File: src/main/java/com/thebluehats/server/core/modules/GlobalTimerModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.game.managers.enchants.GlobalTimer;
import org.bukkit.plugin.java.JavaPlugin;

public class GlobalTimerModule extends AbstractModule {
    @Provides
    @Singleton
    static GlobalTimer provideGlobalTimer(JavaPlugin plugin) {
        return new GlobalTimer(plugin);
    }

    @Override
    protected void configure() {
    }
}

File: minecraft-server/src/main/kotlin/com/thebluehats/server/core/modules/GlobalTimerModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.game.managers.enchants.GlobalTimer
import org.bukkit.plugin.java.JavaPlugin

class GlobalTimerModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        @JvmStatic
        fun provideGlobalTimer(plugin: JavaPlugin): GlobalTimer {
            return GlobalTimer(plugin)
        }
    }
}